src/io/paf.rs

```diff
@@ -0,0 +1,323 @@
+    pub fn records(&mut self) -> Records<R> {
+        Records { inner: self.inner.deserialize() }
+    }
+}
+
+
+/// A PAF record.
+pub struct Records<'a, R: 'a + io::Read> {
+    inner: csv::DeserializeRecordsIter<'a, R, (String, u64, u64, u64, char, String, u64, u64, u64, u64, u64, u8)>,
+}
+
+
+impl<'a, R: io::Read> Iterator for Records<'a, R> {
+    type Item = csv::Result<Record>;
+
+    fn next(&mut self) -> Option<csv::Result<Record>> {
+        self.inner
+            .next()
+            .map(|res| {
+                res.map(|(name_one, length_one, begin_one, end_one, strand, name_second, length_second, begin_second, end_second, base_match, mapping_length, score)| {
+                            Record {
+                            name_one:       name_one,
+                            length_one:     length_one,
+                            begin_one:      begin_one,
+                            end_one:        end_one,
+                            strand:         strand,
+                            name_second:    name_second,
+                            length_second:  length_second,
+                            begin_second:   begin_second,
+                            end_second:     end_second,
+                            base_match:     base_match,
+                            mapping_length: mapping_length,
+                            score:          score,
+                            }
+                        })
+            })
+    }
+}
+
+
+/// A PAF writer.
+pub struct Writer<W: io::Write> {
+    inner: csv::Writer<W>,
+}
+
+
+impl Writer<fs::File> {
+    /// Write to a given file path.
+    pub fn to_file<P: AsRef<Path>>(path: P) -> io::Result<Self> {
+        fs::File::create(path).map(Writer::new)
+    }
+}
+
+
+impl<W: io::Write> Writer<W> {
+    /// Write to a given writer.
+    pub fn new(writer: W) -> Self {
+        Writer {
+            inner: csv::WriterBuilder::new()
+                .has_headers(false)
+                .delimiter(b'\t')
+                .flexible(true)
+                .from_writer(writer),
+        }
+    }
+
+    /// Write a given PAF record.
+    pub fn write(&mut self, record: &Record) -> csv::Result<()> {
+        self.inner.serialize(record)
+    }
+}
+
+
+/// PAF record as defined in "Minimap and miniasm: fast mapping and de novo assembly for noisy long sequences" Heng Li 2017
+/// https://academic.oup.com/bioinformatics/article/32/14/2103/1742895 
+#[derive(Debug, Serialize, Default)]
+pub struct Record {
+    name_one:       String,
+    length_one:     u64,
+    begin_one:      u64,
+    end_one:        u64,
+    strand:    char,
+    name_second:    String,
+    length_second:  u64,
+    begin_second:   u64,
+    end_second:     u64,
+    base_match:     u64, 
+    mapping_length: u64,
+    score:          u8,
+}
+
+impl Record {
+    /// Create a new PAF record.
+    pub fn new() -> Self {
+        Record {
+            name_one:       "".to_owned(),
```

Comment:
I think it should be `first` vs `second` or `one` vs `two`, or `a` vs `b`. Is there an official specification of the format?

Reply:
I make a commit to use `a` vs `b`, near in futur.

The best specifications you can find is probably https://lh3.github.io/minimap2/minimap2.html

Reply:
Thanks. From the spec, it would also make sense to call them query and target, wouldn't it?